fix: link error and add 'world' option for opencv recipe

jim-bcom opened the pull request:
Build used to fail on a link error involving protobuf requesting
missing android logger API: added a patch to fix that.
'world' options enables us to build opencv as one fat library called
opencv_world.

jim-bcom committed:
fix: link error and add 'world' option for opencv recipe

Build used to fail on a link error involving protobuf requesting
missing android logger API: added a patch to fix that.
'world' options enables us to build opencv as one fat library called
opencv_world.

jim-bcom committed:
fix: update package() to handle opencv_world

This allows modules requiring opencv to populate their
conanbuildinfo.pri with the right list of libraries.

granger35 commented:
**command**
conan create . opencv/4.4.0@conan-solar/stable -tf None --build=opencv -o shared=True -s build_type=Release

**configuration:** 
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.cppstd=17
compiler.runtime=MD
compiler.version=15
os=Windows
os_build=Windows
[options]
shared=True
[build_requires]
[env]

**error:**
opencv/4.4.0@conan-solar/stable: ERROR: Package '79bebe36bb0483e24b6f0da9a831d2efb50d3a9a' build failed
opencv/4.4.0@conan-solar/stable: WARN: Build folder C:\.conan\dd4236\1
ERROR: opencv/4.4.0@conan-solar/stable: Error in build() method, line 485
        cmake = self._configure_cmake()
while calling '_configure_cmake', line 391
        cmake.definitions['Protobuf_USE_STATIC_LIBS'] = not self.options['protobuf'].shared
        ConanException: option 'shared' doesn't exist
Possible options are []

granger35 commented:
Tested for Linux gcc (shared=True, gapi=False, gtk=3) and android clang (gapi=False, shared=True, tiff=False), ok for both of them.

jim-bcom committed:
fix: use conan-protobuf, force world on Android, add warning

jim-bcom committed:
fix: propagate shared option to glog (and glfags)

New recipes for these modules have changed default value for 'shared'.
Now, 'shared' is false by default.
This caused an issue when building opencv contrib with shared=True.
The build system looked for shared libs for glflags, didn't find
them and then automatically removed the production of the SFM
module.
But since the recipe did not see this, the package() method
didn't remove sfm from the produced libraries.
This caused the generated compilation/link command lines
to still have sfm, and the linking step failed because the
sfm library was not found.

granger35 committed:
Merge branch 'develop' into fix/opencv-android-logger-link-error
